### PR TITLE
Handle column renaming in makeQueryTemplate and __table

### PR DESCRIPTION
**Pull request by annie**
Implement column renaming in stdlib. Specifically, we now set column name overrides based on the `names` property in `node.data.operations`, which is added in [this PR](https://github.com/observablehq/observablehq/pull/10009).

For SQL data sources, we do this by modifying the `SELECT` clause. For in-memory data sources, we create a copy of the source data and modify it to use the new names.

### CODE REVIEW

**Comment by annie** [comment]
hm looks like CI is failing because `String.replaceAll` (which i added in the unit test) isn't available in Node 14. i'll update it to just use `.replace`.

**Review by mbostock — COMMENTED** [review]
What do you think about making this part of the existing operations.select? E.g.,

```js
operations.select = {
  columns: [
    {column: "foo", as: "foo_alias"},
    {column: "bar", as: "bar_alias"}
  ]
};
```

and then we’d treat the existing operations as if `column` and `as` were the same? I.e., this


```js
operations.select = {
  columns: [
    "foo",
    "bar"
  ]
};
```

would be the same as

```js
operations.select = {
  columns: [
    {column: "foo", as: "foo"},
    {column: "bar", as: "bar"}
  ]
};
```

**Comment by annie** [comment]
oh interesting! in my first pass i had added `names` to `select` as a new property, but that ended up feeling unwieldy – your suggestion seems cleaner though. i'll take a shot at updating it in the monorepo PR and see if i run into any issues.

**Comment by annie** [comment]
@mbostock i ran into one major issue while trying to implement your suggestion, which is that it doesn't allow us to preserve the renamed name when a user hides the column. we could potentially get around this by adding a `hidden` flag to each `columns` object and toggling that when we hide a column, instead of removing the object from the array, but that would require a fairly significant set of changes and i'm not sure it's worth doing.

given that, i think i'd prefer to stick with this `operations.names` approach, unless you have strong objections and/or other ideas!

**Comment by mbostock** [comment]
@annie does that mean we’re still selecting and returning columns that are hidden when the query is run?

**Comment by annie** [comment]
no we don't actually select hidden columns, we just keep the name information for that column in `operations.names`. that allows us to still show the renamed name in the columns dropdown (where hidden columns are listed in a deselected state), and the column will preserve its renamed name if it's reselected.

**Comment by mbostock** [comment]
@annie Got it. Sorry, still waking up! ☕ 👍

**Comment by mbostock** [comment]
No further suggestions from me. I’ll leave the reviewing to @libbey-observable and @mkfreeman! Thanks y’all. 🙏

**Review by libbey-observable — APPROVED** [review]
This all makes sense and looks good to me!

File: src/table.js [hunk 1]
```diff
@@ -659,6 +663,27 @@ export function __table(source, operations) {
       Object.fromEntries(operations.select.columns.map((c) => [c, d[c]]))
     );
   }
+  if (!primitive && operations.names) {
```
Comment (libbey-observable) [inline src/table.js:666]: Ah, nice!

Comment (annie) [reply src/table.js:666]: the real fix for the original issue you found is actually in [this change in Next](https://github.com/observablehq/observablehq/pull/10009/commits/bb017ddd0d86a6e777a420281bd41a4f08feef7e)! but while i was investigating i realized we should also add a check here.

Comment (libbey-observable) [reply src/table.js:666]: aaah! thanks for explaining that.

**Pull request merged**